overlord/hookstate/context.go

```diff
@@ -80,6 +80,11 @@ func (c *Context) SnapRevision() snap.Revision {
 	return c.setup.Revision
 }
 
+// Task returns the task associated with the hook or nil if the context is ephemeral
```

Comment:
This is asking for trouble. I'm willing to bet we'll forget about it somewhere and will have a panicking snapd based on that alone. Let's please have an extra boolean return here which forces people to realize that this may not be set (think _task, ok_ := ...).

It's also slightly unfortunate that we're breaking the abstraction further, but I cannot see any better way.

Reply:
Ok, done.

Agree on remark about Task() getter... I couldn' find any better way of doing that.